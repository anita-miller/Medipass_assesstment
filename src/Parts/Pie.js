import React from 'react'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'
import { CardHeader } from '@material-ui/core';

const Pie = (props) => {
    return(
        <div>  
            {props.store && props.pie ? (
                <Card style={{minWidth: 500}}>
                    <CardHeader 
                        title = {props.store.displayName}
                        subheader={`address : ${props.store.address} , phone : ${props.store.mobile}`}  
                    />
                    <CardContent>               
                        <Typography component="p">
                        {props.pie.displayName} - {props.pie.priceString}
                        </Typography>
                        <Typography component="p">
                            Quantity - {props.pie.quantity}
                        </Typography>
                        <Typography component="p">
                            Rating - {props.store.rating}/10
                        </Typography>
                    </CardContent>
                </Card>
            ): null}
        </div>
    )
}

export default Pie