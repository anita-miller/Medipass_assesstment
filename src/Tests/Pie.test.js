import React from 'react';
import Pie from '../Parts/Pie';
import renderer from 'react-test-renderer';

test('Pie component renders a snapshot', () =>{
    const pie = {
        id: 1,
        storeId: 3,
        displayName: "Beef and curry",
        quantity: 1,
        price: 200,
        priceString: "$2.00",
        isPieOfTheDay: true
        }
    const store = {
        id: 3,
        displayName: "Hungry Jake's",
        address: "345 Bourke Street",
        city: "Melbourne",
        state: "Victoria",
        postcode: "3000",
        mobile: "0412512333",
        coords: {
        latitude: "-37.8130036",
        longitude: "144.9646799"
        },
        rating: 9
        }
    const component = renderer.create(
        <Pie 
            pie={pie}
            store={store}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});