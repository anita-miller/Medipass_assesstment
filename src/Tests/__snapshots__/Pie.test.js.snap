// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pie component renders a snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
    style={
      Object {
        "minWidth": 500,
      }
    }
  >
    <div
      className="MuiCardHeader-root-29"
    >
      <div
        className="MuiCardHeader-content-32"
      >
        <span
          className="MuiTypography-root-35 MuiTypography-headline-40 MuiCardHeader-title-33"
        >
          Hungry Jake's
        </span>
        <span
          className="MuiTypography-root-35 MuiTypography-body1-44 MuiTypography-colorTextSecondary-68 MuiCardHeader-subheader-34"
        >
          address : 345 Bourke Street , phone : 0412512333
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-71"
    >
      <p
        className="MuiTypography-root-35 MuiTypography-body1-44"
      >
        Beef and curry
         - 
        $2.00
      </p>
      <p
        className="MuiTypography-root-35 MuiTypography-body1-44"
      >
        Quantity - 
        1
      </p>
      <p
        className="MuiTypography-root-35 MuiTypography-body1-44"
      >
        Rating - 
        9
        /10
      </p>
    </div>
  </div>
</div>
`;
